lib/ruby/lib/alces/stack/iterator.rb

```diff
@@ -58,11 +38,38 @@ def iterate(gender, lambda, options={})
           Nodes.new(gender).each_with_index do |nodename, index|
             options[:nodename] = nodename
             options[:index] = index
-            output << lambda.call(options)
+            output << Marshal.load(Marshal.dump(lambda.call(options)))
```

Comment:
It creates a deep copy of the return object which could be a hash or an array depending on what is passed in. I was having issues with the return object being altered on subsequent iterations